raiden/raiden_service.py

```diff
@@ -376,7 +376,8 @@ def _callback_new_block(self, current_block_number, chain_id):
             for event in self.blockchain_events.poll_blockchain_events(current_block_number):
                 # These state changes will be procesed with a block_number
                 # which is /larger/ than the ChainState's block_number.
-                on_blockchain_event(self, event, current_block_number, chain_id)
+                # Uses event's block_number, in case it's catching up with old blocks
+                on_blockchain_event(self, event, event.event_data['block_number'], chain_id)
```

Comment:
I'm not sure about this, why do you want to use the event's block number?

Reply:
I use the events block number to avoid calling `on_blockchain_event` with a block number way newer than the event block number being handled now (in the case old events are being queried here, as after a restart).

Reply:
If this is not broken I don't see why it should be changed. This is breaking the contract of `on_blockchain_event` which expects the [`current_block_number` and not the event block number](https://github.com/raiden-network/raiden/blob/master/raiden/blockchain_events_handler.py#L233)

Reply:
Ok, changed it then to fetching the current block number from the client and re-added the setup-time retrieval of events.